bind/log.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Sauce Labs Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package bind
+
+import (
+	"github.com/mmatczuk/anyflag"
+	"github.com/saucelabs/forwarder/httplog"
+)
+
+type httplogFlag struct {
+	*anyflag.SliceValue[NamedParam[httplog.Mode]]
+	update func()
+}
+
+func (f httplogFlag) Set(val string) (err error) {
+	err = f.SliceValue.Set(val)
+
+	if err == nil {
+		f.update()
+	}
+
+	return
+}
+
+func (f httplogFlag) Replace(vals []string) (err error) {
+	err = f.SliceValue.Replace(vals)
+
+	if err == nil {
+		f.update()
+	}
+
+	return
+}
+
+func httplogUpdate(dst, src []NamedParam[httplog.Mode]) {
+	changed := make([]bool, len(dst))
+
+	// Update dst with src values.
+	for i := range dst {
+		for j := range src {
+			if dst[i].Name == src[j].Name {
+				*dst[i].Param = *src[j].Param
+				changed[i] = true
+				break
+			}
+		}
+	}
+
+	// Find default mode if set.
+	defaultMode := httplog.Errors
```

Comment:
How about not setting default mode if it hasn't been found. 
Right now we ignore default modes from config.

Reply:
It's less effective. Now you can set headers and you would get headers everywhere, you can do `headers,api:none` all but api etc.

Reply:
But if you do `--log-http api:body` it would change all the other modes to `httplog.Errors` which is a default in update function. This ignores default values for other modes set there:
```go
bind.HTTPLogConfig(fs, []bind.NamedParam[httplog.Mode]{
		{Name: "api", Param: &c.apiServerConfig.LogHTTPMode},
		{Name: "proxy", Param: &c.httpProxyConfig.LogHTTPMode},
	})
```
Am I missing something?

Reply:
I guess the missing part is that it's irrelevant because the default is Errors.
We could formalize that, remove the default values from DefaultConfigs and make "" resolve to Errors.

Reply:
Or add DefaultMode.

Reply:
It's irrelevant in this case, but it might be a trouble in sc, where we have a different default mode than Errors. 

Allowing user to choose a DefaultMode would be decent.

Reply:
Check the followup comments, now you can set the global default, and the defaulting in bind was removed.